libraries/AP_ExternalControl/AP_ExternalControl.h

```diff
@@ -32,6 +33,16 @@ class AP_ExternalControl
         return false;
     }
 
+    /*
+        Arm the vehicle
+    */
+    virtual bool arm(AP_Arming::Method method, bool do_arming_checks=true);
```

Comment:
Should these be `WARN_IF_UNUSED`? Everything else is in this lib. 

Personally I don't like the default arguments, it can cause confusion. I would wither pass the bool all the way through or just hard code that external control must do the checks. But I'm happy to defer to @Ryanf55 on this.

Reply:
Done, I got distracted by the lack of const in `AP_Arming` and forgot to add it here. 

I removed the defaults and will pass the bools through. We should probably avoid them in use cases like this:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c140-do-not-provide-different-default-arguments-for-a-virtual-function-and-an-overrider